chain/result_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+func TestResultJSON(t *testing.T) {
+	require := require.New(t)
+	errStrBytes := []byte("error")
+	outputBytes := []byte("output")
+	units := fees.Dimensions{1, 2, 3, 4, 5}
+	unitsJSON, err := json.Marshal(units)
+	require.NoError(err)
+	result := Result{
+		Success: true,
+		Error:   errStrBytes,
+		Outputs: [][]byte{outputBytes},
+		Units:   units,
+		Fee:     4,
+	}
+
+	resultJSON, err := json.Marshal(result)
+	require.NoError(err)
+
+	expectedJSON := fmt.Sprintf(`{"error":%q,"fee":4,"outputs":[%q],"success":true,"units":%s}`, codec.Bytes(errStrBytes), codec.Bytes(outputBytes), string(unitsJSON))
```

Comment:
I'd prefer to include the expectedJSON check as well to make sure it's clear that it's returning reasonable JSON output. THis makes it more explicit that we don't have any unexpected values like JSON keys that start with capital letters because there are no annotations.

Certainly may remove this at some point, but I think this is reasonable to include as we are cleaning up the JSON output.